### ISSUE TITLE
Yolo8 Train my own dataset for instance segmentation 

### ISSUE THREAD

**Issue by feitgemel** [ultralytics/ultralytics#7698]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.


### Question

I am trying to train segmentation using Yolo v8 localy.
The data is structed like correctly :

C:/temp/Data-sets/Dog Segmentation/train     -> images
                                                                            -> labels
                                                                            
C:/temp/Data-sets/Dog Segmentation/val     -> images
                                                                            -> labels                                                                            
                                                                            
                                                                            

This is the config.yaml :

path: 'C:/temp/Data-sets/Dog Segmentation'
train: 'train/images' 
val: 'val/images' 

nc: 1
names: ['dog2']

I reduced the train images and labels to 3 in the train and 3 in the validation
There are also 3 annotations file (txt)

Here is an exmaple of 0.txt :

0 0.9306666666666666 0.06012024048096192 0.7253333333333334 0.036072144288577156 0.544 0.13026052104208416 0.232 0.08617234468937876 0.13866666666666666 0.1623246492985972 0.13866666666666666 0.24849699398797595 0.21333333333333335 0.23847695390781562 0.16 0.3026052104208417 0.18666666666666668 0.43286573146292584 0.05333333333333334 0.5170340681362725 0.04533333333333334 0.6332665330661322 0.2613333333333333 0.7394789579158316 0.376 0.7254509018036072 0.45866666666666667 0.9979959919839679 0.6106666666666667 0.9979959919839679 0.6266666666666667 0.8577154308617234 0.736 0.8537074148296593 0.7733333333333333 0.9939879759519038 0.8266666666666667 0.9799599198396793 0.88 0.9138276553106213 0.9973333333333333 0.9118236472945892 0.9973333333333333 0.5210420841683366 0.8666666666666667 0.2665330661322645 0.9813333333333333 0.20641282565130262 0.984 0.10220440881763528


The log has two relevant errors , which are my questions :
1. "WARNING ⚠️ No labels found in C:\temp\Data-sets\Dog Segmentation\train\Images.cache, training may not work correctly"
2. "Scanning C:\temp\Data-sets\Dog Segmentation\train\Images.cache... 0 images, 3 backgrounds, 0 corrupt: 100%|██████████"

Why the 3 images are loaded as backgrounds ?

What can be the problem ?
Eran







### Additional

This is the log :

Ultralytics YOLOv8.1.2 🚀 Python-3.8.18 torch-1.13.1 CUDA:0 (NVIDIA GeForce RTX 3060 Ti, 8191MiB)
WARNING ⚠️ Upgrade to torch>=2.0.0 for deterministic training.
engine\trainer: task=segment, mode=train, model=yolov8l-seg.pt, data=config.yaml, epochs=100, time=None, patience=5, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=0, workers=8, project=c:/temp/Data-sets/dog Segmentation, name=My-Model11, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, 
nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=c:\temp\Data-sets\dog Segmentation\My-Model11
Overriding model.yaml nc=80 with nc=1

                   from  n    params  module                                       arguments
  0                  -1  1      1856  ultralytics.nn.modules.conv.Conv             [3, 64, 3, 2]
  1                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]
  2                  -1  3    279808  ultralytics.nn.modules.block.C2f             [128, 128, 3, True]
  3                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]
  4                  -1  6   2101248  ultralytics.nn.modules.block.C2f             [256, 256, 6, True]
  5                  -1  1   1180672  ultralytics.nn.modules.conv.Conv             [256, 512, 3, 2]
  6                  -1  6   8396800  ultralytics.nn.modules.block.C2f             [512, 512, 6, True]
  7                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]
  8                  -1  3   4461568  ultralytics.nn.modules.block.C2f             [512, 512, 3, True]
  9                  -1  1    656896  ultralytics.nn.modules.block.SPPF            [512, 512, 5]
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 12                  -1  3   4723712  ultralytics.nn.modules.block.C2f             [1024, 512, 3]
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 15                  -1  3   1247744  ultralytics.nn.modules.block.C2f             [768, 256, 3]
 16                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 18                  -1  3   4592640  ultralytics.nn.modules.block.C2f             [768, 512, 3]
 19                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 21                  -1  3   4723712  ultralytics.nn.modules.block.C2f             [1024, 512, 3]
 22        [15, 18, 21]  1   7889779  ultralytics.nn.modules.head.Segment          [1, 32, 256, [256, 512, 512]] 
YOLOv8l-seg summary: 401 layers, 45936819 parameters, 45936803 gradients, 220.8 GFLOPs

Transferred 651/657 items from pretrained weights
Freezing layer 'model.22.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
AMP: checks passed ✅
train: Scanning C:\temp\Data-sets\Dog Segmentation\train\Images.cache... 0 images, **3 backgrounds**, 0 corrupt: 100%|██████████| 3/3 [00:00<?, ?it/s]
**WARNING ⚠️ No labels found** in C:\temp\Data-sets\Dog Segmentation\train\Images.cache, training may not work correctly. See https://docs.ultralytics.com/datasets/detect for dataset f
ormatting guidance.
val: Scanning C:\temp\Data-sets\Dog Segmentation\val\Images.cache... 0 images, 2 backgrounds, 0 corrupt: 100%|██████████| 2/2 [00:00<?, ?it/s]
WARNING ⚠️ No labels found in C:\temp\Data-sets\Dog Segmentation\val\Images.cache, training may not work correctly. See https://docs.ultralytics.com/datasets/detect for dataset for
matting guidance.
Plotting labels to c:\temp\Data-sets\dog Segmentation\My-Model11\labels.jpg... 
zero-size array to reduction operation maximum which has no identity
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically...
optimizer: AdamW(lr=0.002, momentum=0.9) with parameter groups 106 weight(decay=0.0), 117 weight(decay=0.0005), 116 bias(decay=0.0)
100 epochs...

**Comment by feitgemel** [comment]
I uploaded the 3 train images and 3 annotations :

![0](https://github.com/ultralytics/ultralytics/assets/29058350/bae80fc0-8cc2-4b49-97da-2e877147b842)
![1](https://github.com/ultralytics/ultralytics/assets/29058350/8cb75612-afa3-458c-b548-c3917f7cf7d3)
![2](https://github.com/ultralytics/ultralytics/assets/29058350/12022ebd-990d-46b9-9d89-bc1e5366d0c1)

[0.txt](https://github.com/ultralytics/ultralytics/files/13993112/0.txt)
[1.txt](https://github.com/ultralytics/ultralytics/files/13993113/1.txt)
[2.txt](https://github.com/ultralytics/ultralytics/files/13993114/2.txt)

**Comment by feitgemel** [comment]
Here is the train.py :


from ultralytics import YOLO

def main():
    model = YOLO('yolov8l-seg.pt') #    load a pretrained model (recommended for training)
    project = "c:/temp/Data-sets/dog Segmentation"
    experiment = "My-Model" 
    batch_size = 16

    results = model.train(data='config.yaml',  #since we ran it from the main folder, no need for full path
                          epochs=100 , 
                          project=project ,
                          name=experiment, 
                          batch=batch_size,  
                          device=0, 
                          imgsz=640,
                          patience=5,
                          verbose=True,
                          val=True)  




if __name__ == "__main__":
    main()

**Comment by glenn-jocher** [comment]
@feitgemel it appears that your dataset may not be correctly recognized by the training script. Ensure that your label files match the expected format and that the paths in your `config.yaml` are correct. The warning about backgrounds suggests that the images are being read but not associated with any labels. Double-check the structure and content of your annotation files to ensure they are compatible with YOLOv8's expectations for instance segmentation. For further guidance on dataset formatting, please refer to the documentation on segmentation tasks.

**Comment by feitgemel** [comment]
I found the problem

The Yolo v8 train process is case sensitive. 

The images folder name should be in lower case. 

Thanks
Eran

**Comment by glenn-jocher** [comment]
@feitgemel great to hear you've resolved the issue, Eran! Yes, case sensitivity can often be a subtle detail that's easy to overlook. Happy training with YOLOv8! 😊 If you have any more questions or need further assistance, feel free to reach out.
